backend/core/src/main/java/org/sonarsource/sonarlint/core/usertoken/UserTokenService.java

```diff
@@ -45,4 +51,9 @@ public void revokeToken(RevokeTokenParams params, SonarLintCancelMonitor cancelM
       .revoke(params.getTokenName(), cancelMonitor);
   }
 
+  @EventListener
+  public void notifyClientAboutInvalidToken(InvalidTokenEvent event) {
+    client.invalidToken(new InvalidTokenParams(event.getConnectionId()));
```

Comment:
If there is a "connection manager" class as we discussed, maybe it can also take care of notifying the client directly

Reply:
Done.